### PR TITLE
feat: Cascader support cssVar

### PR DESCRIPTION
**Pull request by RedJue**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
- #45618 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Cascader support cssVar      |
| 🇨🇳 Chinese |     级联选择器支持 css 变量      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at a60ec5c</samp>

This pull request refactors and updates the cascader component to use hooks, CSS variables, and the `@ant-design/cssinjs` package for better performance, compatibility, and maintainability. It also aligns the cascader style with the select component and simplifies the SSR logic.

### <samp>🤖 Generated by Copilot at a60ec5c</samp>

*  Import and use `useZIndex` hook to manage z-index of cascader and popups ([link](https://github.com/ant-design/ant-design/pull/45858/files?diff=unified&w=0#diff-4737f6ef564486c449ec019f2826176d8b8d186088e5e912ad06ad1919b2d804R15))
*  Import and use `useSelectCSSVar` hook to register and apply CSS variables for select component ([link](https://github.com/ant-design/ant-design/pull/45858/files?diff=unified&w=0#diff-4737f6ef564486c449ec019f2826176d8b8d186088e5e912ad06ad1919b2d804R29))
*  Rename and move `useStyle` hook to `useCSSVar` and `style` folder for consistency and separation of concerns ([link](https://github.com/ant-design/ant-design/pull/45858/files?diff=unified&w=0#diff-4737f6ef564486c449ec019f2826176d8b8d186088e5e912ad06ad1919b2d804L36-R38))
*  Replace `wrapCascaderSSR` and `wrapSelectSSR` functions with `wrapCascaderCSSVar` and `wrapSelectCSSVar` functions to wrap component node with CSS variables and SSR logic ([link](https://github.com/ant-design/ant-design/pull/45858/files?diff=unified&w=0#diff-4737f6ef564486c449ec019f2826176d8b8d186088e5e912ad06ad1919b2d804L214-R217), [link](https://github.com/ant-design/ant-design/pull/45858/files?diff=unified&w=0#diff-4737f6ef564486c449ec019f2826176d8b8d186088e5e912ad06ad1919b2d804L342-R344))
*  Create `cssVar.ts` file to export function that registers and applies CSS variables for cascader component ([link](https://github.com/ant-design/ant-design/pull/45858/files?diff=unified&w=0#diff-ae8ea15c8edea2bfe3ea0c8c1b345c0eefe24f214370384e77cd0b850a185e2dR1-R4))
*  Import and use `unit` function from `@ant-design/cssinjs` package to convert numbers to CSS units ([link](https://github.com/ant-design/ant-design/pull/45858/files?diff=unified&w=0#diff-9d96d7e729614395a22c393efc9b29ee933f5b16d67972bab03f5c99898a9ceeL1-R1), [link](https://github.com/ant-design/ant-design/pull/45858/files?diff=unified&w=0#diff-9d96d7e729614395a22c393efc9b29ee933f5b16d67972bab03f5c99898a9ceeL62-R62), [link](https://github.com/ant-design/ant-design/pull/45858/files?diff=unified&w=0#diff-8a22481590f9bcba223f0f520b4dbe3b39d1e425b089aceb25c79ba64a3b7371L1-R1), [link](https://github.com/ant-design/ant-design/pull/45858/files?diff=unified&w=0#diff-8a22481590f9bcba223f0f520b4dbe3b39d1e425b089aceb25c79ba64a3b7371L16-R16))

### CODE REVIEW

**Comment by MadCcc** [comment]
截图 diff 有点多，需要看看

**Comment by RedJue** [comment]
> 截图 diff 有点多，需要看看

我看看是不是哪里没改对

**Comment by MadCcc** [comment]
Diff 还是很多

File: components/cascader/index.tsx [hunk 1]
```diff
@@ -211,8 +212,9 @@ const Cascader = React.forwardRef<CascaderRef, CascaderProps<any>>((props, ref)
 
   const rootPrefixCls = getPrefixCls();
 
-  const [wrapSelectSSR, hashId] = useSelectStyle(prefixCls);
-  const [wrapCascaderSSR] = useStyle(cascaderPrefixCls);
+  const [, hashId] = useSelectStyle(prefixCls);
+  const wrapSelectCSSVar = useSelectCSSVar(prefixCls);
+  const wrapCascaderCSSVar = useCSSVar(cascaderPrefixCls);
```
Comment (MadCcc) [inline components/cascader/index.tsx:217]: 这两个要给到 rootCls，弹层组件的 dom 不止一处

Comment (RedJue) [reply components/cascader/index.tsx:217]: 好

File: components/cascader/index.tsx [hunk 2]
```diff
@@ -211,8 +213,10 @@ const Cascader = React.forwardRef<CascaderRef, CascaderProps<any>>((props, ref)
 
   const rootPrefixCls = getPrefixCls();
 
-  const [wrapSelectSSR, hashId] = useSelectStyle(prefixCls);
-  const [wrapCascaderSSR] = useStyle(cascaderPrefixCls);
```
Comment (MadCcc) [inline components/cascader/index.tsx:215]: 少了 这个 useStyle 吧

File: components/cascader/index.tsx [hunk 3]
```diff
@@ -211,8 +213,10 @@ const Cascader = React.forwardRef<CascaderRef, CascaderProps<any>>((props, ref)
...
+  const [, hashId] = useSelectStyle(prefixCls);
+  const rootCls = useCSSVarCls(prefixCls);
+  const wrapSelectCSSVar = useSelectCSSVar(rootCls);
+  const wrapCascaderCSSVar = useCSSVar(rootCls);
```
Comment (MadCcc) [inline components/cascader/index.tsx:219]: 这里的写法参考下 tree-select 吧

Comment (RedJue) [reply components/cascader/index.tsx:219]: 行 我看看

**Review by MadCcc — APPROVED** [review]

**Pull request merged**